Add name from content tests for CSS text-transform:full-size-kana

adampage opened the pull request:
Closes web-platform-tests/interop-accessibility#137.

adampage committed:
remove obsolete comment

adampage committed:
initial full-size-kana tests

adampage commented:
Hi @cookiecrook, I saw that there was already a distinct `name_from_content.tentative.html` file in the repo, so I globbed on to it. Would it be better to isolate these tests in a new file?

I created 2 subtests for this PR. The first subtest is a straightforward `h3` heading that checks its name from contents.

The second subtest is a `<ruby>` construction. This seemed appropriate at first since `full-size-kana` was evidently [conceived for `ruby` annotations](https://github.com/web-platform-tests/interop/issues/558#:~:text=text%2Dtransform%3A%20full%2Dsize%2Dkana%20transforms%20the%20size%20of%20small%20Kana%20characters%20in%20Japanese%20to%20look%20like%20their%20full%2Dsize%20counterparts%20%E2%80%94%20a%20technique%20sometimes%20used%20to%20make%20ruby%20text%20more%20legible%20at%20small%20sizes.) in the first place? However, all 3 of the elements used there [have no ARIA role](https://www.w3.org/TR/html-aam-1.0/#el-rp) and therefore [do not support name from content](https://www.w3.org/TR/wai-aria-1.3/#namefromprohibited), so I suppose this subtest is just straight-up invalid. 😅 Do you think this case is worth testing, and is there some other technique I could use?

adampage commented:
[Chrome and Firefox are both passing the first subtest](https://wpt.fyi/results/accname/name/comp_name_from_content.tentative.html?sha=ff940ac08c&label=pr_head).

cookiecrook reviewed (commented):
editorial clarity suggestions. thanks.

adampage committed:
expand subtests from individual words to sentences, test ruby accname via button element

adampage committed:
fix typos

adampage commented:
Thanks, @cookiecrook. I’ve added sentence context for both subtests:

1. **びょういんのかんじゃサービス** (_Hospital patient services_)
  in which the transformation of びょういん (_hospital_) to びよういん (_beauty parlor_) is unambiguously wrong.
  
2. **竜の背に乗る** (_Ride the dragon‘s back_)
  in which the furigana for 竜 is transformed from りゅう (_dragon_) to りゆう (_reason_), also plainly wrong.

I also updated the second subtest to enclose the `<ruby>` in a button, and then tested the accname of that button.

I’m still waiting for all the results to come through, but it’s notable in Safari that the accname of the button ignores the `<rp>` elements but **does** roll up the `<rt>` elements in addition to the kanji they’re annotating. This results in a confusing label of `竜りゆうの背せに乗のる` that is both duplicative (it loosely translates to “_Ride ride the dragon reason’s back back_” 😅)  and also allows `full-size-kana` to transform the AT-exposed text.

cookiecrook commented:
> Hi @cookiecrook, I saw that there was already a distinct `name_from_content.tentative.html` file in the repo, so I globbed on to it. Would it be better to isolate these tests in a new file?

Yes, I would do it the way you did it.

> The second subtest is a `<ruby>` construction. […] 😅 Do you think this case is worth testing, and is there some other technique I could use?

I also initially thought that would be a good test, but a) it's not a name-able element as you've discovered, and b) ruby is ambiguous wrt to labels, so perhaps these subtests should use something simple like a link or a button.

cookiecrook commented on accname/name/comp_name_from_content.tentative.html:
```diff
@@ -37,6 +37,23 @@ <h3 data-expectedlabel="image link" data-testname="heading with link referencing
+<h3 lang="ja" data-expectedlabel="びょういんのかんじゃサービス" data-testname="heading name from content with text-transform:full-size-kana" class="ex" style="text-transform:full-size-kana;">びょういんのかんじゃサービス</h3>
+
+<!-- “竜の背に乗る” means “Ride the dragon’s back” -->
+<!-- text-transform:full-size-kana incorrectly changes 竜’s <rt> furigana: -->
```
"incorrectly" may be confusing... As it's the correct behavior for rendered text, but should not be transformed in the test exposed to the Accessibility APIs… Note: I expect implementors may have questions about how to achieve this, so it's good this is in a tentative file.

adampage replied:
Here’s [another cut at that](https://github.com/web-platform-tests/wpt/pull/47535/files#diff-974c66e8b6fda730bf40c4e0b0ae3750f4dd71a89e5ee74f0b3dcb10f36900bcR42). I also went ahead and commented out [the second subtest](https://github.com/web-platform-tests/wpt/pull/47535/commits/4d5410bfe7eb96849a4261ca95d8560146374930).

adampage committed:
clarify comments

adampage committed:
comment out ruby subtest

cookiecrook commented on accname/name/comp_name_from_content.tentative.html:
```diff
@@ -37,6 +37,27 @@ <h3 data-expectedlabel="image link" data-testname="heading with link referencing
+<!-- from りゅう (ryū = dragon) to りゆう (riyuu = reason) -->
+<!-- TODO: Restore this subtest when ruby’s accname computation and -->
+<!-- text-to-speech requirements are settled. -->
+<!--
+<h1>link containing a <code>ruby > rt</code> element which visually transforms <span lang="ja">りゅう</span> (dragon) to <span lang="ja">りゆう</span> (reason) using CSS <code>text-transform:full-size-kana</code></h1>
+<button lang="ja" data-expectedlabel="竜の背に乗る" data-testname="ruby > rt name from content with text-transform:full-size-kana" class="ex" >
+  <ruby>竜<rp>(</rp><rt style="text-transform:full-size-kana;">りゅう</rt><rp>)</rp>の背<rp>(</rp><rt>せ</rt><rp>)</rp>に乗<rp>(</rp><rt>の</rt><rp>)</rp>る</ruby>
+</button>
+-->
```
Ruby will require more than a single subtest, so let's keep this test but reformat it to avoid the current problems with testing AccName from Ruby. Something like this?

```suggestion
<button lang="ja" data-expectedlabel="りゅう" data-testname="button with text-transform: full-size-kana on contents" class="ex" style="text-transform: full-size-kana;">りゅう</button>
```

cookiecrook commented on accname/name/comp_name_from_content.tentative.html:
```diff
@@ -37,6 +37,27 @@ <h3 data-expectedlabel="image link" data-testname="heading with link referencing
+<!-- “竜の背に乗る” means “Ride the dragon’s back” -->
+<!-- text-transform:full-size-kana visually changes the <rt> furigana for 竜 -->
+<!-- from りゅう (ryū = dragon) to りゆう (riyuu = reason) -->
+<!-- TODO: Restore this subtest when ruby’s accname computation and -->
+<!-- text-to-speech requirements are settled. -->
```
```suggestion
```

cookiecrook commented on accname/name/comp_name_from_content.tentative.html:
```diff
@@ -37,6 +37,27 @@ <h3 data-expectedlabel="image link" data-testname="heading with link referencing
+<h3 lang="ja" data-expectedlabel="びょういんのかんじゃサービス" data-testname="heading name from content with text-transform:full-size-kana" class="ex" style="text-transform:full-size-kana;">びょういんのかんじゃサービス</h3>
+
+<!-- “竜の背に乗る” means “Ride the dragon’s back” -->
+<!-- text-transform:full-size-kana visually changes the <rt> furigana for 竜 -->
+<!-- from りゅう (ryū = dragon) to りゆう (riyuu = reason) -->
```
```suggestion
<!-- text-transform:full-size-kana visually changes the furigana for 竜 -->
<!-- from りゅう (ryū = dragon) to りゆう (riyuu = reason) -->
<!-- but the computed accessibility label should retain the original meaning. -->
```

adampage committed:
Update accname/name/comp_name_from_content.tentative.html

Co-authored-by: James Craig <cookiecrook@users.noreply.github.com>

adampage committed:
Update accname/name/comp_name_from_content.tentative.html

Co-authored-by: James Craig <cookiecrook@users.noreply.github.com>

adampage committed:
Update accname/name/comp_name_from_content.tentative.html

Co-authored-by: James Craig <cookiecrook@users.noreply.github.com>

adampage committed:
add subtest heading and clarify comments

adampage committed:
remove second subtest
